### PR TITLE
fix: Do not lowercase commit- and pr- title

### PR DESCRIPTION
**Pull request by kingcrunch**
Right now, when semantic commits are enabled, commit- and pr-titles will be lowercased.

- This is not specified, neither by conventional-commits, nor angular. Latter defines, that the first letter should be lowercased, but not the entire title.
- Prevents one from making custom changes. While it is easily possible to adjust the titles to all lowercase with filters and such, it is not possible the other way around, when the titles are lowercased as the very last step.

## Changes

Do not change the case of the titles (PR and commit) anymore.

This does _not_ (yet) cover the [concerns mentioned in the discussion](https://github.com/renovatebot/renovate/discussions/20931#discussioncomment-5317495)!

## Context

- https://github.com/renovatebot/renovate/discussions/20931

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
Unfortunately these changes don't make sense for the users who _don't_ have semantic commits enabled

**Comment by KingCrunch** [comment]
Hi @rarkins 

Can You explain to me, _why_?

The entire "lowercase"-logic seems to be exclusive to the semantic-commit code, because the property `upgrade.toLowerCase` is only set, when `upgrade.semanticCommits === 'enabled'`

```
if (upgrade.semanticCommits === 'enabled' && !upgrade.commitMessagePrefix) {
  // ...
  upgrade.toLowerCase =
        regEx(/[A-Z]/).exec(upgrade.semanticCommitType!) === null &&
        !upgrade.semanticCommitType!.startsWith(':');
}
```

So `upgrade.toLowercase` can _never_ be `true`, when not in semantic-commit mode. So the users without this aren't affected.

What did I miss?

**Comment by rarkins** [comment]
It appears to me that you have modified these to lower case statically. Won't they affect all users?

image

**Comment by KingCrunch** [comment]
Ah, that's what you meant. Yes, in regular-mode this doesn't add any benefit, but imo it doesn't hurt neither.

I am open for suggestions.

My issue now is:
- Right now nobody can decide on their own, if they want lowercase, or not.
- I think, that it makes most sense to keep it up to the user, how they would their commits look like. This includes, that it should look like exactly how it is configured (including defaults/preset)
- And now I struggle how to introduce auto-magic for the angular users into the default and/or presets...


As far as it concerns me I would just keep the uppercase letter. But regarding the discussion it sounded like the angular-approach is something, that should be kept.

**Comment by KingCrunch** [comment]
OK, I tried a different approach: Everythings like usual for regular-mode users, uppercase actions for standard semantic-commit-mode users and opt-in via preset `:angularTitles` for angular setups.

File: lib/config/presets/internal/default.ts [hunk 1]
```diff
@@ -2,6 +2,57 @@ import type { Preset } from '../types';
 
 /* eslint sort-keys: ["error", "asc", {caseSensitive: false, natural: true}] */
 export const presets: Record<string, Preset> = {
+  angularTitles: {
```
Comment (rarkins) [inline lib/config/presets/internal/default.ts:5]: I'm not sure so many of the fields are required here?

Comment (KingCrunch) [reply lib/config/presets/internal/default.ts:5]: Actually I am not sure either.

That are mostly options, which themself are defined as set of multiple options, like

https://docs.renovatebot.com/configuration-options/#pindigest

grafik

I had the feeling to define (e.g.) `pinDigest` to be angular compatible I have to copy all properties, which are defined there.

Comment (rarkins) [reply lib/config/presets/internal/default.ts:5]: I thought you are only modifying commitMessage Action

Comment (KingCrunch) [reply lib/config/presets/internal/default.ts:5]: Yes, exactly. That should be the only difference to the default settings.

Can we maybe speed it a little bit up and you just tell me, what I should do?

Asking me I would just drop the lowercase-logic and let angular-style-fans set the values on their own. If thats fine, the very first commit already solved the issue...

**Comment by rarkins** [comment]
@KingCrunch let's go back to #20931 to discuss requirements. I will close this for now as I'm not sure we'll reuse the code. It can be reopened if I'm wrong

File: lib/config/presets/internal/default.ts (hunk 1 above)
Comment (KingCrunch) [reply lib/config/presets/internal/default.ts:5]: Revert

**Pull request closed without merging**